calico_containers/tests/unit/datastore_test.py

```diff
@@ -1028,46 +1037,190 @@ def test_add_bgp_peer(self):
+                              {"as_num": 32245, "ip": "192.168.100.5"})
+            data["write"] = True
 
+        self.etcd_client.write.side_effect = mock_write
         peer = IPAddress("192.168.100.5")
-        self.datastore.add_bgp_peer("v4", peer)
-        self.etcd_client.write.assert_called_once_with(BGP_PEERS_PATH,
-                                                       "192.168.100.5",
-                                                       append=True)
+        self.datastore.add_bgp_peer("v4", peer, 32245)
+        assert_true(data["write"])
 
-    def test_add_bgp_peer_exists(self):
+    def test_del_bgp_peer_exists(self):
         """
-        Test adding an IP peer when the peer already exists.
+        Test del_bgp_peer() when the peer does exist.
         :return: None
         """
+        self.etcd_client.delete = Mock()
+        peer = IPAddress("192.168.3.1")
+        self.datastore.remove_bgp_peer("v4", peer)
+        # 192.168.3.1 has a key ...v4/0 in the ordered list.
+        self.etcd_client.delete.assert_called_once_with(
+                                                BGP_PEERS_PATH + "192.168.3.1")
 
-        self.etcd_client.read.side_effect = mock_read_2_peers
+    def test_del_bgp_peer_doesnt_exist(self):
+        """
+        Test del_bgp_peer() when the peer does not exist.
+        :return: None
+        """
+        self.etcd_client.delete.side_effect = EtcdKeyNotFound()
+        peer = IPAddress("192.168.100.1")
+        assert_raises(KeyError, self.datastore.remove_bgp_peer, "v4", peer)
 
-        peer = IPAddress("192.168.3.1")
-        self.datastore.add_bgp_peer("v4", peer)
-        assert_false(self.etcd_client.write.called)
+    def test_get_node_bgp_peer(self):
+        """
+        Test getting IP peers from the datastore when there are some peers.
+        :return: None
+        """
+        self.etcd_client.read.side_effect = mock_read_2_node_peers
+        peers = self.datastore.get_node_bgp_peers("TEST_HOST", "v4")
+        assert_equal(len(peers), 2)
+        assert_dict_equal(peers[0], {"ip": IPAddress("192.169.3.1"),
+                                     "as_num": 32245})
+        assert_dict_equal(peers[1], {"ip": IPAddress("192.169.5.1"),
+                                     "as_num": 32245})
 
-    def test_del_bgp_peer_exists(self):
+    def test_get_node_bgp_peer_no_key(self):
+        """
+        Test getting IP peers from the datastore when the key doesn't exist.
+        :return: None
+        """
+        self.etcd_client.read.side_effect = mock_read_2_node_peers
+        peers = self.datastore.get_node_bgp_peers("BLAH", "v4")
+        assert_list_equal([], peers)
+
+    def test_get_node_bgp_peer_no_peers(self):
+        """
+        Test getting BGP peers from the datastore when the key is there but has
+        no children.
+        :return: None
+        """
+        self.etcd_client.read.side_effect = mock_read_no_bgppeers
+        peers = self.datastore.get_bgp_peers("v4")
```

Comment:
test is called `get_node_bgp_peers_no_peers` but you're calling `get_bgp_peers` instead of `get_node_bgp_peers`.